injector/cpu_pressure_test.go

```diff
@@ -98,27 +100,46 @@ var _ = Describe("Failure", func() {
 	})
 
 	Describe("injection", func() {
-
-		It("should join the cpu and cpuset cgroups for the unstressed core", func() {
-			cgroupManager.AssertCalled(GinkgoT(), "Join", "cpu", 666, false)
-			cgroupManager.AssertCalled(GinkgoT(), "Join", "cpuset", 666, false)
-			cgroupManager.AssertNumberOfCalls(GinkgoT(), "Join", 2)
-		})
-
-		It("should prioritize the current process", func() {
-			manager.AssertCalled(GinkgoT(), "Prioritize")
-		})
-
-		It("should run the stress on one core", func() {
-			stresser.AssertNumberOfCalls(GinkgoT(), "Stress", 1)
-		})
-
-		It("should record core and StresserPID in StresserManager", func() {
-			stresserManager.AssertCalled(GinkgoT(), "TrackCoreAlreadyStressed", 1, 666)
+		Context("user request to stress all the cores", func() {
+			BeforeEach(func() {
+				stresserManager.On("TrackInjectorCores", mock.Anything, mock.Anything).Return(cpuset.NewCPUSet(0, 1), nil)
+			})
+
+			It("should join the cpu and cpuset cgroups for the unstressed core", func() {
+				cgroupManager.AssertCalled(GinkgoT(), "Join", "cpu", 666, false)
+				cgroupManager.AssertCalled(GinkgoT(), "Join", "cpuset", 666, false)
+				cgroupManager.AssertNumberOfCalls(GinkgoT(), "Join", 2)
+			})
+
+			It("should prioritize the current process", func() {
+				manager.AssertCalled(GinkgoT(), "Prioritize")
+			})
+
+			It("should run the stress on one core", func() {
+				stresser.AssertNumberOfCalls(GinkgoT(), "Stress", 1)
+			})
+
+			It("should record core and StresserPID in StresserManager", func() {
+				stresserManager.AssertCalled(GinkgoT(), "TrackCoreAlreadyStressed", 1, 666)
+			})
+
+			It("should skip a target core that was already stress", func() {
+				stresserManager.AssertNotCalled(GinkgoT(), "TrackCoreAlreadyStressed", 0, mock.Anything)
+			})
 		})
 
-		It("should skip a target core that was already stress", func() {
-			stresserManager.AssertNotCalled(GinkgoT(), "TrackCoreAlreadyStressed", 0, mock.Anything)
+		Context("user request to stress half of the cores", func() {
+			BeforeEach(func() {
+				userRequestCount := intstr.FromString("50%")
+				spec = v1beta1.CPUPressureSpec{
+					Count: &userRequestCount,
+				}
+				stresserManager.On("TrackInjectorCores", mock.Anything, &userRequestCount).Return(cpuset.NewCPUSet(0, 1), nil)
+			})
+
+			It("should call stresserManager track cores and get new core to apply pressure", func() {
+				// left empty for AfterEach
```

Comment:
Assertions of this `It` are only checked by `AssertExpectations` correct? I would rephrase the comment, at first I did not directly grasped why it was left empty 😅  
```suggestion
				// left empty as AfterEach 'AssertExpectations' check all this tests expectations
```